kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/api/KieServerOpenShift.java

```diff
@@ -40,7 +40,9 @@ default boolean isDCStable(DeploymentConfig dc) {
         List<DeploymentConfig> deployments = client.deploymentConfigs()
                 .withLabel(CFG_MAP_LABEL_SERVER_ID_KEY, serverId).list().getItems();
         if (deployments.isEmpty()) { return Optional.empty();}
-        if (deployments.size() == 1) { return Optional.ofNullable(deployments.get(0)); }
+        if (deployments.size() == 1) {
+            return deployments.stream().filter(dc -> dc.getSpec().getReplicas().intValue() > 0).findFirst();
```

Comment:
@rhtevan Can you please add test coverage making sure that both cases are covered (0 replicas, more replicas)?

Reply:
@sutaakar Added two test cases covering zero and more replicas

Reply:
@rhtevan , @sutaakar , is this change necessary before https://github.com/jboss-container-images/jboss-kie-modules/pull/251 and https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/307 can get merged?

Reply:
@errantepiphany This PR has impact the effectiveness of lifecycle hooks added from this PR (https://github.com/jboss-container-images/jboss-kie-modules/pull/251). It needs to be merged first.